### PR TITLE
Fix a force ssl redirection bug that occur when session store disabled.

### PR DESCRIPTION
**Pull request by supercaracal**
### Summary

https://github.com/rails/rails/pull/19682#issuecomment-261835411

I couldn't request [the same modification](https://github.com/rails/rails/pull/19682/files) as the last time. Because it was degraded at API mode.

@spastorino @prathamesh-sonpatki @rafaelfranca 

### Related Issues
* https://github.com/rails/rails/issues/19679
* https://github.com/rails/rails/issues/27009

### Reproduced apps
* https://github.com/supercaracal/session_store_less_sample
* https://github.com/bronson/force-ssl-repro-app

### CODE REVIEW

**Comment by supercaracal** [comment]
> PG::TRDeadlockDetected: ERROR:  deadlock detected
> SQLite3::BusyException: database is locked: DELETE FROM "delayed_jobs"

They're not seem to be related. Can I rerun this CI?

**Comment by prathamesh-sonpatki** [comment]
Restarted the build, we can ignore that failure if it happens again.

File: actionpack/test/controller/force_ssl_test.rb [hunk 1]
```diff
@@ -92,6 +92,22 @@ def cheeseburger
   end
 end
 
+class RedirectToSSLIfSessionStoreDisabled < ForceSSLController
```
Comment (rafaelfranca) [inline actionpack/test/controller/force_ssl_test.rb]: This is testing an implementation detail. Can we test it at integration level so we don't need to mock the request object?

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: actionpack/test/controller/force_ssl_test.rb (hunk 1 above)
Comment (supercaracal) [inline actionpack/test/controller/force_ssl_test.rb]: I changed at da1453e @rafaelfranca

Comment (supercaracal) [inline actionpack/test/controller/force_ssl_test.rb]: Build [failed](https://travis-ci.org/rails/rails/jobs/178503726). But we can ignore that.

> PG::TRDeadlockDetected: ERROR:  deadlock detected

**Comment by rafaelfranca** [comment]
Backported in c1f990c

**Pull request merged**